OpenSim/Common/DataTable.h

```diff
@@ -0,0 +1,1142 @@
+  ///
+  /// \throws SimTK::Exception::IndexOutOfRange If the row index specified
+  ///                                           is not in the DataTable_.
+  /// \throws OpenSim::ColumnDoesNotExist If the col label specified is not in
+  ///                                     the DataTable_.
+  ET& updElt(const size_t row, const string& collabel) {
+    try {
+      return data.updElt(static_cast<int>(row), 
+                         static_cast<int>(col_ind.at(collabel)));
+    } catch (std::out_of_range exc) {
+      std::string expl{"Column label '" + collabel + "' does not exist."};
+      throw ColumnDoesNotExist{expl};
+    }
+  }
+
+  /// Get a *copy* of the underlying matrix of the DataTable_.
+  std::unique_ptr<SimTK::Matrix_<ET>> getAsMatrix() const {
+    return std::unique_ptr<SimTK::Matrix_<ET>>{new SimTK::Matrix_<ET>{data}};
+  }
+
+  /// Add(append) a row to the DataTable_ using a SimTK::RowVector_. If the 
+  /// DataTable is empty, input row will be the first row. This function can be
+  /// used to populate an empty DataTable_.
+  ///
+  /// \param row The row to be added as a SimTK::RowVector_.
+  ///
+  /// \throws OpenSim::ZeroElements If number of elements in the input row is
+  ///                               zero.
+  /// \throws OpenSim::InvalidEntry If number of columns in the input row does
+  ///                               not match the number of columns of the 
+  ///                               DataTable_.
+  void addRow(const SimTK::RowVector_<ET>& row) {
+    if(row.nrow() == 0 || row.ncol() == 0)
+      throw ZeroElements{"Input row has zero length."};
+    if(data.ncol() > 0 && row.size() != data.ncol())
+      throw InvalidEntry{"Input row has incorrect number of columns."};
+
+    data.resizeKeep(data.nrow() + 1, row.ncol());
+    data.updRow(data.nrow() - 1).updAsRowVector() = row;
+  }
+
+  /// Add(append) a row to the DataTable_ using an iterator(satisfying 
+  /// requirements of an input_iterator) producing one entry at a time. If this
+  /// function is called on an empty DataTable_ without providing ncol_hint, it
+  /// performs <i>allocation + relocation</i> for [log2(ncol) + 1] times where 
+  /// ncol is the actual number of elements produced by the input iterator. To
+  /// add multiple rows at once using an input-iterator, use addRows().
+  ///
+  /// \param first Beginning of range covered by the iterator.
+  /// \param last End of the range covered by the iterator.
+  /// \param ncol_hint Hint for the number of columns in the input iterator. 
+  ///                  Can be approximate above or below the actual number.
+  ///                  This is only used when this function is called on an 
+  ///                  empty DataTable_. Ignored otherwise. Providing a hint
+  ///                  reduces the number of resize operations which involves
+  ///                  memory allocation and relocation.
+  /// \param allow_missing Allow for missing values. When set to false, this
```

Comment:
Sure. I can change it.
I thought the variables are named all lower case with underscores. Or they will look like function names though function names are often verbs.